core/shared/src/main/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLogger.scala

```diff
@@ -52,12 +53,26 @@ object PagingSelfAwareStructuredLogger {
    * @return
    *   SelfAwareStructuredLogger with pagination.
    */
-  def withPaging[F[_]: Monad](pageSizeK: Int = 64, maxPageNeeded: Int = 999)(
+  def withPaging2[F[_]: Monad: UUIDGen](pageSizeK: Int = 64, maxPageNeeded: Int = 999)(
```

Comment:
Sorry, I wasn't feeling creative with naming at this moment 😅 I will happily change this to any suggestion, thanks!

Reply:
Maybe we should name it `paged`? 🤔 and just keep that as the future API

Reply:
Works for me! :)